code/turf/turf_flock.dm

```diff
@@ -356,6 +356,19 @@
 		else if(!isfeathertile(newloc))
 			F.end_floorrunning()
 
+/turf/simulated/wall/auto/feather/Bumped(AM)
+	. = ..()
+	if(istype(AM, /mob/living/critter/flock/drone))
+		var/mob/living/critter/flock/drone/F = AM
+		if(F.floorrunning || (F.can_floorrun && F.resources >= 1))
+			if(!F.client || (F.client && F.client.check_key(KEY_RUN))) //ai doesn't have to hold shift to wallrun, people do
+				F.start_floorrunning()
+				F.set_loc(src)
```

Comment:
```suggestion
				F.set_loc(src)
				F.start_floorrunning()
```
Not sure if it matters, but feels bit off saying floorrun on turf that may not be floorrunnable/not the turf you are floorrunning on

Reply:
it needs to be in that order because `set_loc` triggers `Entered`, which has bonus floorrunning behaviour